### PR TITLE
add workaround for N/A bug in kineto

### PR DESCRIPTION
**Pull request by guotuofeng**
Add workaround for N/A which already fixed in libkineto but not in next release.

### CODE REVIEW

File: tb_plugin/torch_tb_profiler/profiler/data.py [hunk 1]
```diff
@@ -66,7 +66,22 @@ def parse(run_dir, worker):
             # Kineto may export json file with non-ascii code. before this is fixed, use a workaround
```
Comment (nbcsm) [inline tb_plugin/torch_tb_profiler/profiler/data.py:66]: is N/A case still `json.decoder.JSONDecodeError`?

File: tb_plugin/torch_tb_profiler/profiler/data.py [hunk 2]
```diff
@@ -66,7 +66,22 @@ def parse(run_dir, worker):
...
+                    # TODO: remove the workaround after the libkineto fix for N/A is merged into pytorch
+                    if os.path.splitext(trace_path)[1].lower() != ".json":
+                        # ignore gz files
+                        raise
+
+                    f.seek(0)
+                    patched_json = trace_path + "_fixed.json"
+                    with open(patched_json, "wt") as fout:
+                        for line in f:
+                            fout.write(line.replace('N/A', '\"N/A\"'))
```
Comment (nbcsm) [inline tb_plugin/torch_tb_profiler/profiler/data.py:81]: maybe more strict?

File: tb_plugin/torch_tb_profiler/profiler/data.py [hunk 3]
```diff
@@ -66,7 +66,22 @@ def parse(run_dir, worker):
             # Kineto may export json file with non-ascii code. before this is fixed, use a workaround
             # to handleJSONDecodeError, re-encode it and save to a temp file
             with fopen(trace_path, 'r') as f:
-                trace_json = json.load(f, strict=False)
+                try:
+                    trace_json = json.load(f, strict=False)
+                except json.decoder.JSONDecodeError:
+                    # TODO: remove the workaround after the libkineto fix for N/A is merged into pytorch
+                    if os.path.splitext(trace_path)[1].lower() != ".json":
```
Comment (gaoteng-git) [inline tb_plugin/torch_tb_profiler/profiler/data.py:73]: ".gz" seems also works.

File: tb_plugin/torch_tb_profiler/profiler/data.py (hunk 1 above)
Comment (guotuofeng) [reply tb_plugin/torch_tb_profiler/profiler/data.py:66]: yes

**Pull request closed without merging**